app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/CommunityFragment.kt

```diff
@@ -1,8 +1,258 @@
+    private lateinit var menuButton: ImageButton
+    private lateinit var detailsLayout: LinearLayout
+    private lateinit var tournamentsView: RecyclerView
+
+    private lateinit var tournament: TournamentModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        drawer = view.findViewById(R.id.fragment_community)
+        menuButton = view.findViewById(R.id.community_menu_button)
+        detailsLayout = view.findViewById(R.id.community_detail_layout)
+        tournamentsView = view.findViewById(R.id.display_community_tournaments_view)
+
+        tournament = TournamentModel(sampleWeekly(), sampleYourTournaments(), sampleDiscoveryTournaments())
+        if (arguments?.getSerializable("tournaments") != null) {
+            tournament = arguments?.getSerializable("tournaments") as TournamentModel
+        }
+
+        createNavigationMenu(view)
+
+        tournamentsView.layoutManager = LinearLayoutManager(context)
+        tournamentsView.adapter = CommunityTournamentPostViewAdapter(getAllPostsFromAll(), true)
+
+        menuButton.setOnClickListener {
+            drawer.open()
+        }
+
+        //TODO add logic to the sort button
+        //val sortButton = view.findViewById<ImageButton>(R.id.community_sort_button)
+
+        val backButton = view.findViewById<ImageButton>(R.id.community_back_button)
+        backButton.setOnClickListener {
+            menuButton.visibility = View.VISIBLE
+            detailsLayout.visibility = View.GONE
+            tournamentsView.adapter = CommunityTournamentPostViewAdapter(getAllPostsFromAll(), true)
+            tournamentsView.invalidate()
+        }
+
+    }
+
+    /**
+     * create the navigation menu where the different tournaments are displayed
+     */
+    private fun createNavigationMenu(view: View) {
+        val menu = view.findViewById<NavigationView>(R.id.community_navigation_view).menu
+
+        val weekly = menu.addSubMenu("Weekly tournament")
+        weekly.add(tournament.getWeeklyTournament().name)
+            .setContentDescription("${tournament.getWeeklyTournament().name} details")
+            .setOnMenuItemClickListener {
+                menuItemListener(view, sampleWeekly())
+            }
+
+        val your = menu.addSubMenu("Your tournament")
+        for (i in tournament.getYourTournament("placeholder")) {
+            your.add(i.name)
+                .setContentDescription("${i.name} details")
+                .setOnMenuItemClickListener {
+                    menuItemListener(view, i)
+                }
+        }
+        val discover = menu.addSubMenu("Discover")
+        for (i in tournament.getDiscoverTournament("placeholder")) {
+            discover.add(i.name)
+                .setContentDescription("${i.name} details")
+                .setOnMenuItemClickListener {
+                    menuItemListener(view, i)
+                }
+        }
```

Comment:
These three block of codes are very similar, it should be possible to factorize them with a private function.

Reply:
It is now inside a private function